### ISSUE TITLE
Incorrect `ValidationError` for non-executed operations in multiple-operations documents

### ISSUE THREAD

**Issue by cairomassimo** [graphql-rust/juniper#455]
**Describe the bug**
Calling `juniper::execute` fails with a `ValidationError` when a multiple-operation document is given and the provided variables do not validate for *any* of the operations defined in the document, even if they are *not* executed.

Use Case: multiple-operations documents ared used to generate client code using `graphql-client generate` (they are especially convenient for many small-size operations). In this case, the whole original document is sent every time an operation is performed.

**To Reproduce**

Running the following code fails with `ValidationError([RuleError { locations: [SourcePosition { index: 22, line: 0, col: 22 }], message: "Variable \"$x\" of required type \"String!\" was not provided." }])', src/libcore/result.rs:1084:5`:
```rust
extern crate juniper;

use juniper::RootNode;
use std::collections::HashMap;

struct Query;
struct Mutation;

#[juniper::object]
impl Query {
    fn y(x: Option<String>) -> i32 { 0 }
}

#[juniper::object]
impl Mutation {}

type Schema = RootNode<'static, Query, Mutation>;

fn main() {
    juniper::execute(
        "query A { y } query B($x: String!) { y(x: $x) }",
        Some("A"),
        &Schema::new(Query, Mutation),
        &HashMap::new(),
        &(),
    ).unwrap();
}
```

**Expected behavior**
It should run the operation `A` successfully, as it does not define the variable `$x`.

Per:
https://graphql.github.io/graphql-spec/draft/#sec-Executing-Requests
only the variables defined by the executed operation are validated.
> Let coercedVariableValues be the result of CoerceVariableValues(schema, operation, variableValues).

**Additional context**
Version 0.14.1

**Comment by davidpdrsn** [comment]
I'm still new to the codebase but it seems this isn't possible to do currently.

`execute` returns `Result<(Value<S>, Vec<ExecutionError<S>>), GraphQLError<'a>>`, meaning we either return some data or an error, but not both. The [function that validate the input variables](https://github.com/graphql-rust/juniper/blob/master/juniper/src/validation/input_value.rs#L61) returns `GraphQLError` and and since `execute` returns a `Result` it cannot return both `(Value, Vec<ExecutionError>)` and `GraphQLError` which I guess is needed for this case.

**Comment by cairomassimo** [comment]
The issue is with the signature of `validate_input_values`: it should also accept the selected operation.

Say:

```rust
pub fn validate_input_values<S>(
    values: &Variables<S>,
    document: &Document<S>,
    operation_name: Option<&str>,
    schema: &SchemaType<S>,
) -> Vec<RuleError> {
    // [...]
}
```

and something like

```rust
    // [...]
    for def in document {
        if let Definition::Operation(ref op) = *def {
            if Some(ref name) = operation_name {
                if name != op.name { continue; }
            }
            if let Some(ref vars) = op.item.variable_definitions {
                validate_var_defs(values, &vars.item, schema, &mut errs);
            }
        }
    }
```

I may have some time to help with this issue later this week, but I'm also new to the codebase.
I will let you know if I manage to work on this. Please let me know as well if you work on this, so we don't duplicate the work.

**Comment by cairomassimo** [comment]
I forked the repo and hopefully I will soon make a PR fixing this!

**Comment by cairomassimo** [comment]
Any news on this? Should be fixed by #462 but I got no comments on that.
Thanks!

**Comment by LegNeato** [comment]
Just pushed a release with this. Please confirm it is fixed!
